Shopify/quilt
# react-i18n - add `from-dictionary-index` plugin mode

## Description
Adds a `from-dictionary-index` plugin.  With this, folks with lots of CPU (or time) can generate multiple versions of their app.  Each with a specific locale embedded directly into its JS.  The advantages of this are:
* Bundlers like webpack don't get bloated with metadata about where to find async translations
* Users don't have to download multiple sets of translations

## 🎩 
* Check out the `solo-i18n-test` branch of web-rails-proving-ground
* Generate French index files using:
    * `node -e "require('@shopify/react-i18n/generate-dictionaries').generateTranslationDictionaries(['fr'])"`
* Clear out old transpilation artifacts
    * `rm -fr tmp/sewing-kit/cache/webpack`
* `dev run`
* Ensure French translations appear in the `customers` route
* Ensure _only_ French translations are embedded in JS files

<img width="1161" alt="Customers 2019-11-27 15-09-36" src="https://user-images.githubusercontent.com/673655/69756241-021c6200-1128-11ea-9efb-a91fe31ce046.png">

## Type of change

- [x] `@shopify/react-i18n Minor: New feature

## Checklist

- [x] I have added a changelog entry, prefixed by the type of change noted above
